rmw_connext_cpp/src/rmw_request.cpp

```diff
@@ -28,19 +28,16 @@ rmw_send_request(
   const void * ros_request,
   int64_t * sequence_id)
 {
-  if (!client) {
-    RMW_SET_ERROR_MSG("client handle is null");
-    return RMW_RET_ERROR;
-  }
+  RMW_CHECK_ARGUMENT_FOR_NULL(client, RMW_RET_INVALID_ARGUMENT);
+
   RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
     client handle,
     client->implementation_identifier, rti_connext_identifier,
-    return RMW_RET_ERROR)
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
 
-  if (!ros_request) {
-    RMW_SET_ERROR_MSG("ros request handle is null");
-    return RMW_RET_ERROR;
-  }
+  RMW_CHECK_ARGUMENT_FOR_NULL(ros_request, RMW_RET_INVALID_ARGUMENT);
+
+  RMW_CHECK_ARGUMENT_FOR_NULL(sequence_id, RMW_RET_INVALID_ARGUMENT);
 
   ConnextStaticClientInfo * client_info = static_cast<ConnextStaticClientInfo *>(client->data);
   if (!client_info) {
```

Comment:
@Lobotuerk We can reduce LOC count a bit here. This check and the two following ones make no sense IMHO, as those imply that:

- the caller is not using the API outside the spec, or
- the implementation is logically incorrect, or
- the process memory space is corrupt.

Reply:
Should this be added into another PR, as I don't think completely relates to this "Error Update" PR